src/node_data.cpp

```diff
@@ -256,9 +256,9 @@ void node_data::reset_map() {
 }
 
 void node_data::insert_map_pair(node& key, node& value) {
-  m_map[&key] = &value;
+  m_map.emplace_back(kv_pair(&key, &value));
```

Comment:
I don't understand what you're saying

```
emplace_back(kv_pair(&key, &value))
```

and

```
emplace_back(&key, &value)
```

don't compile to the same thing. The former uses pair's copy-constructor and the latter uses pair's two-arg constructor.